src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -20,9 +20,10 @@
   "Creates the routes to handle the master's '/v3' routes, which
    includes '/environments' and the non-CA indirected routes. The CA-related
    endpoints are handled separately by the CA service."
-  [request-handler
+  [request-handler-fn
    get-code-id-fn]
-  (let [request-handler-with-code-id (request-core/wrap-with-code-id request-handler get-code-id-fn)]
+  (let [request-handler (comp request-handler-fn request-core/wrap-params-for-jruby)
+        request-handler-with-code-id (request-core/wrap-with-code-id request-handler-fn get-code-id-fn)]
```

Comment:
@haus, @KevinCorcoran It feels a little strange to me to have `master_core` be coupled directly to `request-core` like this.  One other option that occurs to me is that we could just have this layer add something to the request map - maybe just an `add-code-id-to-params: {boolean}` or something - but then have the logic that actually calls back on the versioned code service to get the code id if necessary, etc.., could all just be done down inside of the request handler service's `jruby-request-handler`.

A potentially related question, does it matter when the service function to get the code-id is called relative to when the request reserves the jruby instance that it will be using to perform the catalog request?  If the code-id is added to the request as this PR currently has it, it seems like the Ruby code running the catalog request could be changed out between the time that the code-id is computed and the catalog compilation takes place - i.e., due to a file sync update being done in between.  Could that mean that the effective code-id related to the content on disk might be different than that supplied with the request down into the Ruby code?  In that case, doing the request for the code-id down in the `jruby-request-handler` could ensure that no file sync update would occur while the code-id is being computed - because the file sync update would be held off while the JRuby instance is held.

I don't have much of a handle on what code-id means at this point, so very possible that my comments here are all garbage and can be ignored.  :)